spec/requests/application_controller_spec.rb

```diff
@@ -143,6 +143,21 @@
         expect(response.status).to eq(404)
         expect(response.body).to_not include('google.com/search')
       end
+
+      it 'should cache results' do
+        $redis.set('not_found_page_topics', 'first response')
+        get '/t/nope-nope/99999999'
+        expect(response.status).to eq(404)
+        expect(response.body).to include('first response')
+
+        SiteSetting.login_required = true
+        sign_in(Fabricate(:user))
+
+        $redis.set('not_found_page_topics', 'second response')
```

Comment:
We need to ensure that the key in Redis is cleaned up at the end of the test to ensure that it doesn't leak the state to other tests that follow.